contracts/Octobay.sol

```diff
@@ -344,20 +361,132 @@ contract Octobay is Ownable, ChainlinkClient, BaseRelayRecipient {
-                issueDeposits[_depositId].amount,
-            'This issue deposit has been withdrawn already.'
-        );
-        payable(msg.sender).transfer(issueDeposits[_depositId].amount);
+        require(issueStatusByIssueId[issueDeposits[_depositId].issueId] == IssueStatus.OPEN, 'Issue is not OPEN.');
+
+        uint256 payoutAmt = issueDeposits[_depositId].amount;
         issueDepositsAmountByIssueId[
             issueDeposits[_depositId].issueId
-        ] -= issueDeposits[_depositId].amount;
+        ] -= payoutAmt;
+        emit RefundIssueDepositEvent(msg.sender, payoutAmt, issueDeposits[_depositId].issueId, _depositId);
         delete issueDeposits[_depositId];
+        payable(msg.sender).transfer(payoutAmt);
     }
 
 
 
+    // ------------ ISSUE CLAIMING ------------ //
+
+    event WithdrawIssueDepositEvent(string issueId, address recipient, uint256 amount);
+
+    struct IssueWithdrawRequest {
+        string issueId;
+        address payoutAddress;
+    }
+
+    mapping(bytes32 => IssueWithdrawRequest) public issueWithdrawRequests;
+
+    function withdrawIssueDeposit(
+        address _oracle,
+        string calldata _issueId
+    ) public oracleHandlesJob(_oracle, 'claim') returns(bytes32 requestId) {
+        require(issueStatusByIssueId[_issueId] == IssueStatus.OPEN, 'Issue is not OPEN.'); 
+
+        (bytes32 jobId, uint256 jobFee) = oracleStorage.getOracleJob(_oracle, 'claim');
+        Chainlink.Request memory request =
+            buildChainlinkRequest(
+                jobId,
+                address(this),
+                this.confirmWithdrawIssueDeposit.selector
+            );
+        request.add('githubUserId', userAddressStorage.userIdsByAddress(msg.sender));
+        request.add('issueId', _issueId);
+        requestId = sendChainlinkRequestTo(_oracle, request, jobFee);
+        issueWithdrawRequests[requestId] = IssueWithdrawRequest({
+            issueId: _issueId,
+            payoutAddress: msg.sender
+        });
+    }
+
+    function confirmWithdrawIssueDeposit(bytes32 _requestId)
+        public
+        recordChainlinkFulfillment(_requestId)
+    {
+        require(issueWithdrawRequests[_requestId].payoutAddress != address(0) );
+        address payoutAddr = issueWithdrawRequests[_requestId].payoutAddress;
+        uint256 payoutAmt = issueDepositsAmountByIssueId[issueWithdrawRequests[_requestId].issueId];
+        issueDepositsAmountByIssueId[issueWithdrawRequests[_requestId].issueId] = 0;
+        issueStatusByIssueId[issueWithdrawRequests[_requestId].issueId] = IssueStatus.CLAIMED;
+        emit WithdrawIssueDepositEvent(issueWithdrawRequests[_requestId].issueId, payoutAddr, payoutAmt);
+        delete issueWithdrawRequests[_requestId];
+        // delete issueDeposits[_depositId]; ??? loop through issueDepositIdsByIssueId ???
+        // awardGovernanceTokens(payoutAddr, payoutAmt, tokenAddr); // How do we go from issueId => tokenAddr ?
```

Comment:
@mktcode - this is something that needs a bit of thought...

A few options:

* We use an Oracle request to look up the repo and org associated with the issue ID. We can then either award whatever tokens are associated with either of these.
* We add a function which is protected by Owner NFT which sets the token address for a particular issue.

Maybe both?